### ISSUE TITLE
Frontend O11y / APM mapping: Allow overriding the service name through the Faro SDK

### ISSUE THREAD

**Issue by cedricziel** [grafana/faro-web-sdk#749]
## Description

As a user of Frontend O11y and Application O11y in Grafana Cloud, i would like to override the service name 
for a subset of telemetry as the user progresses through the UX.

Example:
* User lands on the login page, the telemetry should be mapped to the `login` service
* User progresses through the app, goes to checkout, the associated telemetry should be associated with  the `checkout` service
* User manages their newsletter preferences, associated telemetry should be tagged with the `profile` service name

The override should be persisted in the user session to allow maximum flexibility.

## Proposed solution

Leverage the `setView` API and allow to override the context.

Example:

```
setView('login', {'service_override': 'login'});

# all telemety generated herafter, will be transmitted with a service_override meta that the cloud receiver can use

setView('checkout', {'service_override': 'checkout'});
```

**Comment by Catchpowle** [comment]
@cedricziel I think this would work great for us, but could I just clarify that this would apply both to the Span created when using the Tracing Instrumentation with Faro and the standard session-related logs, such as errors?

My only other thought was whether you'd want to provide a more generic solution for adding/overriding attributes as part of this API, but for us, I think this would be enough as it is.

If we wanted to add/update more attributes, they'd likely be to add info such as the name of the owning team, which can be derived by knowing the service name anyway. Therefore, having a `service_override` meta set here would mean we'd be able to make use of it in the `onBeforeSend` hook for any other service-related attribute modification we might want, as far as I understand. 🙂

**Comment by cedricziel** [comment]
> I think this would work great for us, but could I just clarify that this would apply both to the Span created when using the Tracing Instrumentation with Faro and the standard session-related logs, such as errors?

The traces will have the updated service.name attribute, yes. We should probably think about enabling overrides for service.namespace and service.version as well.

In my book the conversion here would be done by us in cloud (overriding the service.name attribute) and it would apply to all signals produced after the sdk method call.

**Comment by cedricziel** [comment]
> Therefore, having a service_override meta set here would mean we'd be able to make use of it in the onBeforeSend hook for any other service-related attribute modification we might want, as far as I understand. 🙂

Yes - you can use that hook to inspect the full structure and make decisions based on it

**Comment by Catchpowle** [comment]
@cedricziel This all sounds good, thank you for prioritising it!

**Comment by cedricziel** [comment]
@Catchpowle we have a PR up for Faro in #893 ; backend change pending

**Comment by codecapitano** [comment]
@Catchpowle the feature is released and you can no override the service.name.
Docs are currently in review but let me share the very raw version here to get you started.

```

# Override the service name attribute

Faro provides APIs to dynamically change the service name, as the user navigates across your app.
The service name attributes in the logs and traces are immediately updated by the Faro receiver.

This feature is for example useful when multiple teams own parts of the product.
Instead of using multiple isolated Faro instances running in the same browsing context, you can just
change the service name. 

The following properties  are updated in the telemetry:

- `service.name` resource attribute on traces and spans
- `resource_service_name` in logs

## How it works

When triggering a service name change Faro updates the session metadata with an overrides object
which contains the new service name. The Faro receiver reads the instruction and updates the service
name for all subsequent signals till another service name is set.
The service name change persists across page loads.

## How to change a service name

Changing the service name can be done by either by calling the `setView()` and `setSession()` APIs.
Both APIs have a second `options` parameter which is used to instruct Faro change the service name.

**Using the `setView()` API **

Change view name and service name:

```ts
faro.api.setView({ name: 'my-new-view' }, { overrides: { serviceName: 'my-new-view-service' } });
```

**Using the `setSession()` API **

Change session metadata and service name:

```ts
faro.api.setSession(mySessionMeta, { overrides: { serviceName: 'my-new-view-service' } });
```

If the sessions meta parameter is empty, Faro creates a new session and adds the overrides to that session.

```ts
faro.api.setSession(undefined, { overrides: { serviceName: 'my-new-view-service' } });
```
```
